### PR TITLE
Feature/custom editor field with file upload

### PR DESCRIPTION
**Pull request by ra-external**
- Add custom TinyMCE editor field to AdminUI
- create REST API to upload images for this field

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -141,6 +145,15 @@ temp/
 # Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
 
 # User-specific stuff
+tsout
+tsout/**
+tsout/**/admin
+_deprecated
+.deprecated/*
+.idea/keystone-blank-client.iml
+.idea
```
Comment (amorote) [inline .gitignore]: It's not working properly because there are .idea/ files commited

File: lists/Article.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+				type: Wysiwyg,
+
+				hooks: {
+					resolveInput({ resolvedData, existingItem }) {
+						// TODO -- there must be a better way to do this?
+						// TinyMCE returns to KeystoneJS all the host paths replaced with '../../'
+						// Get rid of that and replace with the proper host string
+						// Before saving to the DB
+						const hostString = getHostString();
+						console.log('resolvedData');
```
Comment (amorote) [inline lists/Article.ts]: Try to remove console logs before commiting

File: lists/Image.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { File } from '@keystonejs/fields';
+
+
+export default (keystone, { imageFileAdapter }) => {
+	console.log('Image', imageFileAdapter);
+
+	keystone.createList('Image', {
+		schemaDoc: 'Image for Article',
+		labelResolver: item => item?.file?.originalFilename ?? '',
```
Comment (amorote) [inline lists/Image.ts]: How does it work? I've never seen ??

File: .gitignore (hunk 1 above)
Comment (ra-external) [reply .gitignore]: Do you know how to remove them once it's been commited?

File: lists/Article.ts (hunk 2 above)
Comment (ra-external) [reply lists/Article.ts]: Vale, lo haré

Comment (ra-external) [reply lists/Article.ts]: Adelante

File: lists/Image.ts (hunk 3 above)
Comment (ra-external) [reply lists/Image.ts]: ¿Qué?  labelResolver?  Es que puedes gestionar cómo se crean los labels

**Comment by ra-external** [comment]
> Good job!
> I do not think we should upload 'static_files/images', as they are not in each other's database.
 Not sure what the alternative is.  I tried uploading the images directly to the Keystone DB through their GraphQL API but there's some issue -- most likely with their server -- that's preventing that.  I raised the issue but have heard nothing.

File: .gitignore (hunk 1 above)
Comment (amorote) [reply .gitignore]: i guess removing them on local and do a commit

File: lists/Image.ts (hunk 3 above)
Comment (amorote) [reply lists/Image.ts]: I mean the operator '??'. I've never seen it. I've seen '||' to do the same as you are doing

Comment (ra-external) [reply lists/Image.ts]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_Coalescing_Operator

File: .gitignore (hunk 1 above)
Comment (ra-external) [reply .gitignore]: Right, but I don't want to remove the .idea files...I'll figure it out.  I put those in .gitignore the first thing but sometimes .gitignore _ignores_ what I've told it to do

**Pull request merged**